flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/ManuallyTriggeredScheduledExecutorService.java

```diff
@@ -54,6 +57,14 @@
 
     private boolean shutdown;
 
+    public ManuallyTriggeredScheduledExecutorService() {
+        super(DEFAULT_NB_THREADS);
+    }
+
+    public ManuallyTriggeredScheduledExecutorService(int nbThreads, ThreadFactory factory) {
+        super(nbThreads, factory);
```

Comment:
This change IMO doesn't have any effect. Can you please elaborate?

Reply:
except set the exception handler in ManuallyTriggeredScheduledExecutorService

Reply:
since we override the `execute(...)` method (which is the only one used by the coordinator), how exactly do we leverage that?

Reply:
`CoordinatorExecutorThreadFactory#newThread()` sets the exception handler of the created thread.   The JVM calls `Thread#getUncaughtExceptionHandler()` when the threads raises an exception.

Reply:
That part I understand. What I miss is where do we actually create the thread (call the `#newThread` method), since we seem to be executing the `Runnable` passed to the `execute(...)` directly?

Reply:
you're right, the context as a reference to the factory and calls the exception handler directly. No more need for that change, reverted.